pkg/loader/loader.go

```diff
@@ -126,9 +127,24 @@ func readTool(ctx context.Context, prg *types.Program, base *source, targetToolN
 		return loadProgram(data, prg, targetToolName)
 	}
 
-	tools, err := parser.Parse(bytes.NewReader(data))
-	if err != nil {
-		return types.Tool{}, err
+	var tools []types.Tool
+	if strings.HasSuffix(base.Name, ".json") ||
```

Comment:
We should detect if the data is openapi format by inspecting the data itself and not by file extension. That should be more flexible in the future. I also don't want something where the openai schema is hosted at path like "https://example.com/openapi" and it not working becaues it doesn't end with the extension.

Reply:
Fixed